### PR TITLE
kernelci.org: Add build configuration documentation

### PR DESCRIPTION
**Pull request by broonie**
Mostly ported over from the wiki with a little editing.

Signed-off-by: Mark Brown <broonie@kernel.org>

### CODE REVIEW

**Comment by gctucker** [comment]
Deployed on staging: https://static.staging.kernelci.org/docs/builds/builds/

File: kernelci.org/content/en/docs/builds/_index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: "Buildss"
```
Comment (gctucker) [inline kernelci.org/content/en/docs/builds/_index.md:2]:
```suggestion
title: "Builds"
```
Guess that was just a typo?

**Review by gctucker — COMMENTED** [review]
It appears under a Builds / Builds section, I think there's an issue here.  Maybe the page should be under a Configuration section instead? We could have documentation to cover all the YAML config syntax.

**Comment by broonie** [comment]
I put this as a separate section sine the structure seems to work out to having lots of little directories - there's a lot of stuff that could be seen as configuration that's already in separate directories (eg, the secrets stuff in the admin directory is also configuration, and there's some overlap in the LAVA lab bit too). It might be worth merging this as-is and then doing a followup which restructures the current set of documentation into fewer directories?

**Review by gctucker — APPROVED** [review]

**Pull request merged**